src/gui/log/logmodel.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020  Prince Gupta <jagannatharjun11@gmail.com>
+ * Copyright (C) 2019  sledgehammer999 <hammered999@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "logmodel.h"
+
+#include <QApplication>
+#include <QDateTime>
+#include <QColor>
+#include <QPalette>
+
+#include <base/logger.h>
+
+BaseLogModel::BaseLogModel(int initialSize, QObject *parent)
+    : QAbstractListModel(parent)
+    , m_msgCount(initialSize)
+    , m_cache(69)
+{
+}
+
+int BaseLogModel::rowCount(const QModelIndex &) const 
+{
+    return m_msgCount;
+}
+
+int BaseLogModel::columnCount(const QModelIndex &) const
+{
+    return 1;
+}
+
+QVariant BaseLogModel::data(const QModelIndex &index, const int role) const
+{
+    static int hit = 0, miss = 0;
+    if (!index.isValid() || m_msgCount == 0)
+        return {};
+
+    const int i = m_msgCount - index.row() + m_startIndex - 1;
```

Comment:
nope cache size is fixed, it will fill up to the size specified in the BaseLogModel constructor (64 for now) and then start deleting the least recently used

Reply:
We cache no all messages but several recent ones (in current PR implementation).